pcsx2/GS/Renderers/HW/GSRendererHW.cpp

```diff
@@ -1690,6 +1690,38 @@ void GSRendererHW::Draw()
 
 	// The rectangle of the draw
 	m_r = GSVector4i(m_vt.m_min.p.xyxy(m_vt.m_max.p)).rintersect(GSVector4i(context->scissor.in));
+
+	if (!GSConfig.UserHacks_DisableSafeFeatures)
+	{
+		// Constant Direct Write without texture/test/blending (aka a GS mem clear)
+		if ((m_vt.m_primclass == GS_SPRITE_CLASS) && !PRIM->TME // Direct write
+			&& (!PRIM->ABE || m_context->ALPHA.IsOpaque()) // No transparency
+			&& (m_context->FRAME.FBMSK == 0) // no color mask
+			&& !m_context->TEST.ATE // no alpha test
+			&& (!m_context->TEST.ZTE || m_context->TEST.ZTST == ZTST_ALWAYS) // no depth test
+			&& (m_vt.m_eq.rgba == 0xFFFF) // constant color write
+			&& m_r.x == 0 && m_r.y == 0) { // Likely full buffer write
```

Comment:
{ bracket should be on new line.

Reply:
I literally copy/pasted existing code. But I guess I can change it.